app/src/main/java/ch/epfl/sweng/erpa/activities/DiceActivity.java

```diff
@@ -1,152 +1,126 @@
+        flowLayout = findViewById(R.id.dice_layout);
     }
 
-    @OnClick(R.id.rollButton)
-    public void rollDices(View view) {
-        int[] rolls = getNumberOfRolls();
-        if (IntStream.of(rolls).sum() > 15) {
-            createPopup("The number of dice must be less or equal to 15", this);
-        } else {
-            clearAllResults();
-            rollAndShowAllDice(rolls);
-        }
+    //Handle toolbar items clicks
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        Boolean found = onOptionItemSelectedUtils(item.getItemId(), findViewById(R.id.dice_drawer_layout));
+        return found || super.onOptionsItemSelected(item);
     }
 
     /**
-     * Rolls and shows all dice
-     * @param rolls An array containing all the dice to roll
+     * Checks if any die is rolling else rolls all dice
+     * @param view not used
      */
-    private void rollAndShowAllDice(int[] rolls) {
-        int n = 0;
-        for (int i = 0; i < 7; ++i) {
-            String dieType = "D" + dice[i];
-            for (int j = 0; j < rolls[i]; ++j) {
-                allResultViews[n].setText(dieType + ": " + rollDie(dieType));
-                ++n;
+    @OnClick(R.id.rollButton)
+    public void rollDices(View view) {
+        if (Stream.of(allDice).allMatch(dice -> !dice.isRolling())) {
+            for (DieSketch die : allDice) {
+                die.roll();
             }
         }
     }
 
     /**
-     * Rolls one die
-     * @param dieType The type of die to roll
-     * @return The result of the roll
+     * Removes the last die on the layout
+     * @param view
      */
-    private int rollDie(String dieType) {
-        int result = -1;
-        switch (dieType) {
-            case "D4":
-                result = rng.nextInt(4);
-                break;
-            case "D6":
-                result = rng.nextInt(6);
-                break;
-            case "D8":
-                result = rng.nextInt(8);
-                break;
-            case "D10":
-                result = rng.nextInt(10);
-                break;
-            case "D12":
-                result = rng.nextInt(12);
-                break;
-            case "D20":
-                result = rng.nextInt(20);
-                break;
-            case "D100":
-                result = rng.nextInt(100);
-                break;
-            default:
-                throw new AssertionError("Die type is incorrect");
+    @OnClick(R.id.dice_layout)
+    public void removeDie(View view) {
+        if (!allDice.isEmpty()) {
+            int index = allDice.size() - 1;
+            allDice.remove(index);
+            flowLayout.removeViewAt(index);
         }
-        return ++result;
     }
 
     /**
-     * Gets all the TextViews where we write the results of the rolls
-     * @return An array containing all the TextViews
+     * Add a die on the FlowLayout if there is place depending on which button is pressed
```

Comment:
This sentence is weird. ("If there is place left for another die, add the die corresponding to the pressed button" ?)